include/cuco/detail/bloom_filter/bloom_filter_impl.cuh

```diff
@@ -280,80 +270,23 @@ class bloom_filter_impl {
   // memory_to_use, cuda_thread_scope<NewScope> scope = {}) const noexcept;
 
  private:
-  template <class HashValue>
-  __device__ size_type block_idx(HashValue hash_value) const
-  {
-    // TODO use fast_int modulo
-    return hash_value % num_blocks_;
-  }
-
-  // we use the LSB bits of the hash value to determine the pattern bits for each word
-  template <class HashValue>
-  __device__ auto pattern(HashValue hash_value) const
-  {
-    cuda::std::array<word_type, block_words> pattern{};
-    auto constexpr word_bits           = sizeof(word_type) * CHAR_BIT;
-    auto constexpr bit_index_width     = cuda::std::bit_width(word_bits - 1);
-    word_type constexpr bit_index_mask = (word_type{1} << bit_index_width) - 1;
-
-    auto const bits_per_word = pattern_bits_ / block_words;
-    auto const remainder     = pattern_bits_ % block_words;
-
-    uint32_t k = 0;
-#pragma unroll block_words
-    for (int32_t i = 0; i < block_words; ++i) {
-      for (int32_t j = 0; j < bits_per_word + (i < remainder ? 1 : 0); ++j) {
-        if (k++ >= pattern_bits_) { return pattern; }
-        pattern[i] |= word_type{1} << (hash_value & bit_index_mask);
-        hash_value >>= bit_index_width;
-      }
-    }
-
-    return pattern;
-  }
-
-  template <class HashValue>
-  __device__ word_type pattern_word(HashValue hash_value, uint32_t i) const
-  {
-    auto constexpr word_bits           = sizeof(word_type) * CHAR_BIT;
-    auto constexpr bit_index_width     = cuda::std::bit_width(word_bits - 1);
-    word_type constexpr bit_index_mask = (word_type{1} << bit_index_width) - 1;
-
-    auto const bits_per_word = pattern_bits_ / block_words;
-    auto const remainder     = pattern_bits_ % block_words;
-    auto const bits_so_far   = bits_per_word * i + (i < remainder ? i : remainder);
-
-    hash_value >>= bits_so_far * bit_index_width;
-
-    // Compute the word at index i
-    word_type word  = 0;
-    int32_t j_limit = bits_per_word + (i < remainder ? 1 : 0);
-
-    for (int32_t j = 0; j < j_limit; ++j) {
-      word |= word_type{1} << (hash_value & bit_index_mask);
-      hash_value >>= bit_index_width;
-    }
-
-    return word;
-  }
-
   template <uint32_t NumWords>
   __device__ auto vec_load_words(size_type index) const
```

Comment:
nit: I think this function would be easier to understand with either a comment, an explicit return type, or both.

Reply:
Added the explicit return type